docs/security.rst

```diff
@@ -247,3 +247,17 @@ backend. In order to setup an application:
 5. Fill in the required information (the 'Authorization callback URL' must be fully qualifed e.g. http://airflow.example.com/example/ghe_oauth/callback)
 6. Click 'Register application'
 7. Copy 'Client ID', 'Client Secret', and your callback route to your airflow.cfg according to the above example
+
+
+Impersonation
+'''''''''''''
+
+Airflow has the ability to impersonate unix user to while running task
+instances based on the task's ``run_as_user`` parameter. Here are a few
+requirements for this to work properly.
+
+- the unix user needs to exist on the worker (duh), Airflow just assumes that
+  it can ``sudo su`` into that user
+- the user should be able to read the ``airflow.cfg`` file
+- the user needs to be able to write logs in the folder specified in the
```

Comment:
On that configuration transfer as far as I understand the only thing a task really needs are the connections it uses (conn_ids). In https://github.com/apache/incubator-airflow/pull/1391 I played around with grabbing that information from an API - worked pretty well. This is nice because it allows you to fine grain access to those connections. This issue to solve then is how to make sure that someone running "airflow run" xxx cannot access those connection ids. This requires some key passing from the scheduler to task.

This can also be solved (partially) by passing the configuration. However to do that properly and only pass the information the task needs you would need to find out what connections the task requires. On the other hand passing it at run time by the scheduler already solves one issue (you cannot do airflow run xxx as you dont have the config).

I think an issue could lie in backfills and subdagoperators you would need to pass the config here as well, if you want to do that sanitized it might be easier to do it with a key and have API access.

It wont solve the logging issue except when you pass in that config that it should log to a facility other than a file it does not have access to.